wgpu-core/src/command/bind.rs

```diff
@@ -98,29 +98,56 @@ impl BindGroupEntry {
         Provision::Changed { was_compatible }
     }
 
-    pub fn expect_layout(&mut self, bind_group_layout_id: BindGroupLayoutId) -> LayoutChange {
+    pub fn expect_layout<B: GfxBackend>(
+        &mut self,
+        bind_group_layout_id: BindGroupLayoutId,
+        bind_group_layout_storage: &Storage<BindGroupLayout<B>, BindGroupLayoutId>,
+    ) -> LayoutChange {
         let some = Some(bind_group_layout_id);
         if self.expected_layout_id != some {
             self.expected_layout_id = some;
             match self.provided {
                 Some(BindGroupPair {
                     layout_id,
                     ref group_id,
-                }) if layout_id == bind_group_layout_id => {
-                    LayoutChange::Match(group_id.value, &self.dynamic_offsets)
+                }) => {
+                    if layout_id == bind_group_layout_id {
```

Comment:
I think we can do even better than this. What if we resolve the duplicate layouts earlier, in `create_bind_group_layout`? If we see that a layout with the same content already exists, we can just return the same ID. This means we'd always be able to compare by ID.

Reply:
Nice! I'll give it a shot in the coming days 👍

Reply:
Thanks! And sorry for coming up with this late. I had a feeling that we were missing something, but didn't realize what it was until now :sweat_smile: